File: gameserver/src/main/java/l2s/gameserver/network/l2/s2c/ExPutIntensiveResultForVariationMake.java
package l2s.gameserver.network.l2.s2c;

import l2s.gameserver.network.l2.OutgoingExPackets;

public class ExPutIntensiveResultForVariationMake implements IClientOutgoingPacket
{
	private int _refinerItemObjId, _lifestoneItemId, _gemstoneItemId, _unk;
	private long _gemstoneCount;

	public ExPutIntensiveResultForVariationMake(int refinerItemObjId, int lifeStoneId, int gemstoneItemId, long gemstoneCount)
	{
		_refinerItemObjId = refinerItemObjId;
		_lifestoneItemId = lifeStoneId;
		_gemstoneItemId = gemstoneItemId;
		_gemstoneCount = gemstoneCount;
		_unk = 1;
	}

	@Override
	public boolean write(l2s.commons.network.PacketWriter packetWriter)
	{
		OutgoingExPackets.EX_PUT_INTENSIVE_RESULT_FOR_VARIATION_MAKE.writeId(packetWriter);
		packetWriter.writeD(_refinerItemObjId);
		packetWriter.writeD(_lifestoneItemId);
		packetWriter.writeD(_gemstoneItemId);
		packetWriter.writeQ(_gemstoneCount);
		packetWriter.writeD(_unk);

		return true;
	}
}

File: gameserver/src/main/java/l2s/gameserver/network/l2/s2c/augmentation/ExPutIntensiveResultForVariationMake.kt
package l2s.gameserver.network.l2.s2c.augmentation

import l2s.gameserver.network.l2.s2c.L2GameServerPacket

/**
 * @author Java-man
 * @since 21.05.2019
 */
class ExPutIntensiveResultForVariationMake(
    private val refinerItemObjId: Int,
    private val lifestoneItemId: Int,
    private val gemstoneItemId: Int,
    private val gemstoneCount: Long
) : L2GameServerPacket() {

    override fun writeImpl() {
        writeD(refinerItemObjId)
        writeD(lifestoneItemId)
        writeD(gemstoneItemId)
        writeQ(gemstoneCount)
        writeD(1) // TODO unknown
    }

}